flang/test/Lower/call-site-mangling.f90

```diff
@@ -77,3 +77,6 @@ function foo4()
   foo4 = bar4()
 end function
 
+function omp_get_num_threads() bind(c)
```

Comment:
nit: this a test testing the mangling on the callee side. It belongs to `test/Lower/program-units-fir-mangling.f90`.

Could you also add 3 other tests to cover all bind(C) cases, something like:

and in `test/Lower/program-units-fir-mangling.f90`
```
! CHECK-LABEL: func @SomeOtherName()
function somename() bind(c, name="SomeOtherName")
end function
```

in call-site-mangling.f90
```
! CHECK-LABEL: test_bindc
subroutine test_bindc()
  interface
    subroutine somecproc() bind(c)
    end subroutine
    subroutine somecproc2() bind(c, name="anotherName")
    end subroutine
  end interface
  ! CHECK: fir.call @anotherName()
  call somecproc()
  ! CHECK: fir.call @somecproc()
  call somecproc2()
end subroutine
```